java-symbolic-execution/src/main/java/org/sonar/java/se/checks/NonNullSetToNullCheck.java

```diff
@@ -166,7 +169,9 @@ private void checkVariable(CheckerContext context, MethodTree tree, final Symbol
 
   private static boolean isJavaxValidationConstraint(Symbol symbol) {
     SymbolMetadata.AnnotationInstance annotation = symbol.metadata().nullabilityData().annotation();
-    return annotation != null && annotation.symbol().type().fullyQualifiedName().startsWith("javax.validation.constraints.");
+    return annotation != null
+      && (annotation.symbol().type().fullyQualifiedName().startsWith("javax.validation.constraints.")
+      || annotation.symbol().type().fullyQualifiedName().startsWith("jakarta.validation.constraints."));
```

Comment:
I feel like it is starting to become less readable now with 2 different packages, what do you think of something like:

```suggestion
    if (annotation != null) {
      String fullyQualifiedName = annotation.symbol().type().fullyQualifiedName();
      return fullyQualifiedName.matches("^(javax|jakarta)\\.validation\\.constraints\\..*");
    }
    return false;
```

Reply:
I like your simplification! I'll opt for a mixed solution:
`
    return annotation != null
      && annotation.symbol().type().fullyQualifiedName().matches("^(javax|jakarta)\\.validation\\.constraints\\..*"); 
`